File: backend/app/main/models/Transaction.py
from app import db
from datetime import datetime
from sqlalchemy.orm import relationship


class Transaction(db.Model):
    __tablename__ = 'transaction'
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    id_store = db.Column(db.ForeignKey('store.id'), index=True)
    date = db.Column(db.DateTime, default=datetime.utcnow)
    value = db.Column(db.Float)
    cpf = db.Column(db.String(11))
    card = db.Column(db.String(12))
    hour = db.Column(db.DateTime)
    transaction_type = db.Column(db.Integer)

    store = relationship('Store')

File: backend/app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Transaction extends Model
{
    use HasFactory, SoftDeletes;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'import_history_id',
        'type_transaction_id',
        'date',
        'value',
        'cpf',
        'card',
        'hour',
        'store_owner',
        'store_name',
    ];

    /**
     * Find a Import History
     *
     */
    public function importHistory()
    {
        return $this->belongsTo(\App\Models\ImportHistory::class, 'import_history_id');
    }

    /**
     * Find a Type Transaction
     *
     */
    public function typeTransaction()
    {
        return $this->belongsTo(\App\Models\TypeTransaction::class, 'type_transaction_id');
    }


}
